src/App.jsx

```diff
@@ -3,13 +3,20 @@ import "./App.css";
 import { Route, Routes } from "react-router-dom";
 import Homepage from "../pages/Homepage";
 import ViewArticle from "../pages/ViewArticle";
+import TopicCards from "../components/TopicCards";
+import ViewArticleByTopic from "../pages/ViewArticleByTopic";
 
 function App() {
   return (
     <>
       <Routes>
-        <Route path="/" element={<Homepage />}></Route>
-        <Route path="/articles/:article_id" element={<ViewArticle />}></Route>
+        <Route path="/" element={<Homepage />} />
+        <Route path="/articles/:article_id" element={<ViewArticle />} />
+        <Route path="/articles/topics" element={<TopicCards />} />
+        <Route
+          path="/articles/sort/:topic"
```

Comment:
The path you have defined here looks like you are sorting by a topic, but is this actually true? Or are you filtering by a topic?